Data_Engine/Query/Linq/OrderBy.cs

```diff
@@ -0,0 +1,71 @@
+/*
+ * This file is part of the Buildings and Habitats object Model (BHoM)
+ * Copyright (c) 2015 - 2019, the respective contributors. All rights reserved.
+ *
+ * Each contributor holds copyright over their respective contributions.
+ * The project versioning (Git) records all such contribution source information.
+ *                                           
+ *                                                                              
+ * The BHoM is free software: you can redistribute it and/or modify         
+ * it under the terms of the GNU Lesser General Public License as published by  
+ * the Free Software Foundation, either version 3.0 of the License, or          
+ * (at your option) any later version.                                          
+ *                                                                              
+ * The BHoM is distributed in the hope that it will be useful,              
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of               
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the                 
+ * GNU Lesser General Public License for more details.                          
+ *                                                                            
+ * You should have received a copy of the GNU Lesser General Public License     
+ * along with this code. If not, see <https://www.gnu.org/licenses/lgpl-3.0.html>.      
+ */
+
+
+using BH.oM.Reflection.Attributes;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Linq;
+using BHRE = BH.Engine.Reflection;
+
+namespace BH.Engine.Data
+{
+    public static partial class Query
+    {
+        /***************************************************/
+        /**** Public Methods                            ****/
+        /***************************************************/
+
+        [Description("Orders a list of objects by a property of the object - e.g. input 'Name' to sort objects by their names. Note that not all object properties have sortable values")]
+        [Input("objects", "List of objects to be sorted. All objects in the list should be of similar type")]
+        [Input("propertyName", "The name of the property to sort the list of objects by. Note that the property must be able to be sorted - e.g. numbers/text")]
+        public static List<T> OrderBy<T>(this List<T> objects, string propertyName)
+        {
+            if (objects == null || objects.Count == 0) return null;
+            int groupedObjects = objects.GroupBy(x => x.GetType()).Count();
+            System.Type objectType = objects[0].GetType();
```

Comment:
Two things here:
- `objects[0]` can potentially be null. So this would crash
- Am I blind or is it not used anywhere ?

Reply:
No not blind, is a WIP hangover. The problem I was trying to solve was to try and cast/be explicit in object types as at the moment, the specific IComparable methods on objects are not found. The example is a BHoM Point - it has an IComparer<Point>, but as we're using List<T> it doesn't work. Not sure yet how to get around this - @awakeman  and I did a quick hack to explicitly cast the OrderBy Linq method to (Point) and the objects inside, and the IComparer worked, but don't know how to cast to a variable... I'm sure this doesn't make sense, sorry! 

![image](https://user-images.githubusercontent.com/15233608/68453836-a0cf3600-01ee-11ea-9d35-eca8829086fe.png)